controllers/nnf_workflow_controller_helpers.go

```diff
@@ -1246,6 +1246,66 @@ func (r *NnfWorkflowReconciler) waitForContainersToStart(ctx context.Context, wo
 	return nil, nil
 }
 
+func (r *NnfWorkflowReconciler) deleteContainers(ctx context.Context, workflow *dwsv1alpha2.Workflow, index int) (*result, error) {
+	doneMpi := false
+	doneNonMpi := false
+
+	// Set the delete propagation
+	policy := metav1.DeletePropagationBackground
+	deleteAllOptions := &client.DeleteAllOfOptions{
+		DeleteOptions: client.DeleteOptions{
+			PropagationPolicy: &policy,
+		},
+	}
+	// Add workflow matchLabels + directive index (if desired)
+	matchLabels := dwsv1alpha2.MatchingWorkflow(workflow)
+	if index >= 0 {
+		matchLabels[nnfv1alpha1.DirectiveIndexLabel] = strconv.Itoa(index)
+	}
+
+	// Delete MPIJobs
+	mpiJobList, err := r.getMPIJobs(ctx, workflow, index)
+	if err != nil {
+		if strings.Contains(err.Error(), "no kind is registered for the type v2beta1.MPIJobList") || apierrors.IsNotFound(err) {
```

Comment:
I would trim this string to "no kind is registered for the type", just to future-proof it a bit.